domain/port/state/types.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
```

Comment:
Comments on everything

Reply:
I'm not sure if these strictly need doc strings. They aren't exported, and are very short lived within state.go

Reply:
Done